### ISSUE TITLE
Something weird happens at page 2399

### ISSUE THREAD

**Issue by Snigelson** [clubby789/homestuck-downloader#2]
[Page 2399](https://www.homestuck.com/story/2399) does not exist, and if any page after that is requested, the next one is downloaded (i.e. when page 3000 is requested, page 3001 is downloaded).

There must be more such missing pages, as when for example page 8000 is requested, page 8006 is downloaded instead.

Is it safe, if these missing pages are located, to treat them as hard-coded special cases?

**Comment by clubby789** [comment]
Thanks, I'll have a look

**Comment by clubby789** [comment]
I think I know the problem with 2398, or have a vague idea at least. The 'next' clause looks for a link with
`next = soup.find(href="/story/"+str(pageNum+1)).get_text()`. However, as the next page here is actually pageNum+2, it doesn't find it, and crashes the thread, moving onto the next page. I'll add a try: except: there.

**Comment by clubby789** [comment]
And to make matters worse, the image on page 2400 is CALLED 2399.gif

**Comment by clubby789** [comment]
Latest commit fixes errors, but now the text file numbering gets progressively higher than the image numbering. However, 2401.txt *does* contain the text from 2401

**Comment by Snigelson** [comment]
Ahh of course, the next page thing makes sense.

Strange about the image numbering though. I wonder how they have solved that on the official page. At least if one could compile a list of all missing pages this could be compensated for.

**Comment by Snigelson** [comment]
I have a fix in my fork that skips broken pages when looking for next page title.

**Comment by Snigelson** [comment]
I also have a fix for image name compensation now. I have no idea if it makes any sense but I did it anyway.

**Comment by clubby789** [comment]
I think I understand what you're doing there. However, a list of broken pages doesn't seem to be the best option. I think each download should begin by checking the status response of that page; a 404 means we treat it like anything in brokenPages.

**Comment by Snigelson** [comment]
Yeah, I have a more sane solution to the stupid image numbering now :)

Still need a solution for finding the next page link without the list, though.

**Comment by clubby789** [comment]
I'm going to add you as a collaborator, since at this point I'm just merging all your code anyway.
Also, it's probably a safe bet that 2 pages won't be broken in a row, so we could try pageNum + 1?
